src/zimcheck.cpp

```diff
@@ -107,24 +107,25 @@ class ErrorLogger {
         testStatus[type] = status;
     }
 
-    void addError(TestType type, const std::string& message) {
-        errors[type].push_back(message);
+    void addReportMsg(TestType type, const std::string& message) {
+        reportMsgs[type].push_back(message);
     }
 
     void report(bool error_details) const {
-        for (auto status : testStatus) {
-            if (status.second == false) {
-                auto &p = errormapping[status.first];
+        for (auto testmsg : reportMsgs) {
+                auto &p = errormapping[testmsg.first];
                 std::cout << "[" + tagToStr[p.first] + "] " << p.second << ":" << std::endl;
-                for (auto& msg: errors.at(status.first)) {
+                for (auto& msg: testmsg.second) {
                     std::cout << "  " << msg << std::endl;
                 }
-            }
         }
     }
 
     inline bool overalStatus() const {
-        return std::all_of(testStatus.begin(), testStatus.end(), [](std::pair<TestType, bool> e){ return e.second == true;});
+        return std::all_of(testStatus.begin(), testStatus.end(),
+                           [](std::pair<TestType, bool> e){
+                                    return errormapping[e.first].first == LogTag::ERROR && e.second == true;
```

Comment:
This should be `errormapping[e.first].first != LogTag::ERROR || e.second`

return true (pass) if this is not a error test or if this is a error test and the test passes.

Reply:
thanks. fixed